File: 01_Language/01_Functions/python/http_build_query.py
# coding: utf-8

import urllib.parse

def http_build_query(query_data):
    return urllib.parse.urlencode(query_data)

if __name__ == '__main__':
    d = {
        "foo" : "bar",
        "baz" : "boom",
        "cow" : "milk",
        "php" : "hypertext processor",
        0 : ["a", "b", "c"],
    }
    print(http_build_query(d))

File: 01_Language/01_Functions/php/http_build_query.php
<?php

$data = array(
    'foo' => 'bar',
    'baz' => 'boom',
    'cow' => 'milk',
    'php' => 'hypertext processor',
    ["a", "b", "c"]
);

echo http_build_query($data) . "\n";
echo http_build_query($data, '', '&amp;') . "\n";


function _format_postkey($post, &$result, $key = null) {
    foreach($post as $k => $v) {
        $_k = $key !== null ? $key.'['.$k.']' : $k;
        if(is_array($v)) {
            _format_postkey($v, $result, $_k);
        } else {
            $result[$_k] = $v;
        }
    }
}
_format_postkey($data, $result);
var_dump($result);
echo http_build_query($result) . "\n";
